worker/upgradesteps/manifold.go

```diff
@@ -52,11 +51,11 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 			if err := context.Get(config.AgentName, &agent); err != nil {
 				return nil, err
 			}
-
-			// Grab the tag and ensure that it's for a machine.
-			tag, ok := agent.CurrentConfig().Tag().(names.MachineTag)
-			if !ok {
-				return nil, errors.New("agent's tag is not a machine tag")
+			switch agent.CurrentConfig().Tag().(type) {
```

Comment:
@axw,
Do you think that this check (and similar one in the worker) still needed? 
I mean with this PR, upgradesteps is run on agent, does not matter if it's unit or machine... And we only have unit or machine agent :D

Reply:
good point - drop it

Reply:
done